cms/admin/pageadmin.py

```diff
@@ -75,6 +75,7 @@
 from cms.utils.admin import jsonify_request
 from cms.utils.conf import get_cms_setting
 from cms.utils.urlutils import admin_reverse
+from cms.utils.log_operations import log_page_change, log_page_move, log_page_delete
```

Comment:
I'm foreseeing issues with the implementations of title, plugins, etc because this is too tied to the admin.
As a result, I think it might be a better approach to use signal handlers and listen for the post_obj_operation signals for pages and post_placeholder_operation signal for placeholders and plugins.

I suggest to create a module called log_entries.py in cms/signals/
and there add all the necessary handlers.

So this won't really change much in your current implementation, other than where the log entry is created.

Reply:
All of this is fine but falls short for title changes as the signals for these are attached to a page change. No signals are fired specifically to title. One way to achieve this would be to catch a page edit and check what was being changed and log dependant on that. It's not a clean solution.